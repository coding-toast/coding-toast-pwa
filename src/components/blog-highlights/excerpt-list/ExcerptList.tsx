import * as React from 'react';
import { Row, Col } from 'react-bootstrap';
import BasicCard from '../../card/BasicCard';

interface IExcerptListProps {
  blogs: IBlog[];
}

interface IBlog {
  banner: string;
  blogId: string;
  description: string;
  publishDate: string;
  title: string;
}

const ExcerptList: React.FC<IExcerptListProps> = (props) => {
  const { blogs } = props;
  return (
    <Row as='section' xs={1} md={2}>
      {blogs?.map((post: IBlog) => {
        const { title, banner, description, publishDate, blogId } = post;
        return (
          <Col key={blogId}>
            <BasicCard title={title} banner={banner} description={description} publishDate={publishDate} />
          </Col>
        );
      })}
    </Row>
  );
};
export default ExcerptList;
