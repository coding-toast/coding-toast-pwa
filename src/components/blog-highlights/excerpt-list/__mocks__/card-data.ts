export const query = [
  {
    blogId: '968769ef-231d-5501-8942-ee51d8d1324d',
    title: 'blog-post-placerholder1',
    publishDate: '`January 1, 1970`',
    description: 'first blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  },
  {
    blogId: '1a172731-99f4-5f68-8ead-23324a44f70c',
    title: 'blog-post-placerholder3',
    publishDate: '`January 1, 1970`',
    description: 'third blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  },
  {
    blogId: '011af6f6-6c3f-5eab-81ad-3f73e427dbd8',
    title: 'blog-post-placerholder4',
    publishDate: '`January 1, 1970`',
    description: 'fourth blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  },
  {
    blogId: 'd28f2392-be5f-5fc1-96fb-b3c69e7dce6b',
    title: 'blog-post-placerholder2',
    publishDate: '`January 1, 1970`',
    description: 'second blog post',
    banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
  }
];
