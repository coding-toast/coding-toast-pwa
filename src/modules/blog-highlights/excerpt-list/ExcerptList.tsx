import * as React from 'react'
import { Row, Col } from 'react-bootstrap'
import { CardBody, CardTitle, CardDescription, CardMeta } from '../../../components/card'
import { ButtonWithIcon } from '../../../components/buttons/button-with-icon/ButtonWithIcon'

interface ExcerptListProps {
  allMarkdownRemark: {
    allMarkdownRemark: {
      edges: Post[]
    }
  }
}
interface Post {
  node: {
    id: string
    frontmatter: {
      banner: string
      description: string
      title: string
      publishDate: string
    }
  }
}
const ExcerptList: React.FC<ExcerptListProps> = (props) => {
  const { allMarkdownRemark } = props
  return (
    <Row as="section" xs={1} md={2}>
      {allMarkdownRemark.allMarkdownRemark.edges.map((post: Post) => {
        const { title, banner, description, publishDate } = post.node.frontmatter
        const { id } = post.node
        return (
          <Col>
            <CardBody key={id}>
              <CardTitle title={title} backgroundUrl={banner} />
              <CardDescription text={description} />
              <CardMeta publishDate={publishDate} />
              <ButtonWithIcon toUrl="/" />
            </CardBody>
          </Col>
        )
      })}
    </Row>
  )
}
export default ExcerptList
