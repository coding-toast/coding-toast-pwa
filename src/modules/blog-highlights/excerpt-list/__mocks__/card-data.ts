export const query = {
  allStrapiBlogs: {
    edges: [
      {
        node: {
          blogId: '968769ef-231d-5501-8942-ee51d8d1324d',
          title: 'blog-post-placerholder1',
          publishDate: '`January 1, 1970`',
          description: 'first blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: '68e1f043-9989-584f-80e7-2bb2eed3f985',
          title: 'blog-post-placerholder5',
          publishDate: '`January 1, 1970`',
          description: 'fifth blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: '1a172731-99f4-5f68-8ead-23324a44f70c',
          title: 'blog-post-placerholder3',
          publishDate: '`January 1, 1970`',
          description: 'third blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: '011af6f6-6c3f-5eab-81ad-3f73e427dbd8',
          title: 'blog-post-placerholder4',
          publishDate: '`January 1, 1970`',
          description: 'fourth blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      },
      {
        node: {
          blogId: 'd28f2392-be5f-5fc1-96fb-b3c69e7dce6b',
          title: 'blog-post-placerholder2',
          publishDate: '`January 1, 1970`',
          description: 'second blog post',
          banner: 'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg'
        }
      }
    ]
  }
}
