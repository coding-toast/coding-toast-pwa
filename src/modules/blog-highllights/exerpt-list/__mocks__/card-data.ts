export const query = {
  allMarkdownRemark: {
    edges: [
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'second blog post',
            title: 'blog-post-placerholder2',
            publishDate: 'June 16, 2007'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'third blog post',
            title: 'blog-post-placerholder3',
            publishDate: 'January 30, 2001'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'fourth blog post',
            title: 'blog-post-placerholder4',
            publishDate: 'February 13, 1995'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'fifth blog post',
            title: 'blog-post-placerholder5',
            publishDate: 'December 7, 2005'
          }
        }
      },
      {
        node: {
          frontmatter: {
            author: 'author here',
            banner:
              'https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg',
            description: 'first blog post',
            title: 'blog-post-placerholder1',
            publishDate: 'December 29, 2015'
          }
        }
      }
    ]
  }
}
