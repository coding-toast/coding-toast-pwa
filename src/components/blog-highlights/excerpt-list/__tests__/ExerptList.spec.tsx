import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ExcerptList from '../ExcerptList';
import { query } from '../__mocks__/card-data';

it('renders correctly', () => {
  const component = shallow(<ExcerptList blogs={query} />);
  expect(toJson(component)).toMatchSnapshot();
});
