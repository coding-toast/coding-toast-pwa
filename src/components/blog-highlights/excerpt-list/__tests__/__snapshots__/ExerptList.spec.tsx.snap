// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Row
  as="section"
  md={2}
  noGutters={false}
  xs={1}
>
  <Col
    key="968769ef-231d-5501-8942-ee51d8d1324d"
  >
    <BasicCard
      banner="https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg"
      description="first blog post"
      publishDate="\`January 1, 1970\`"
      title="blog-post-placerholder1"
    />
  </Col>
  <Col
    key="1a172731-99f4-5f68-8ead-23324a44f70c"
  >
    <BasicCard
      banner="https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg"
      description="third blog post"
      publishDate="\`January 1, 1970\`"
      title="blog-post-placerholder3"
    />
  </Col>
  <Col
    key="011af6f6-6c3f-5eab-81ad-3f73e427dbd8"
  >
    <BasicCard
      banner="https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg"
      description="fourth blog post"
      publishDate="\`January 1, 1970\`"
      title="blog-post-placerholder4"
    />
  </Col>
  <Col
    key="d28f2392-be5f-5fc1-96fb-b3c69e7dce6b"
  >
    <BasicCard
      banner="https://cliparting.com/wp-content/uploads/2016/06/Ribbon-outline-banner-clip-art-free-vector-for-free-download-about.jpg"
      description="second blog post"
      publishDate="\`January 1, 1970\`"
      title="blog-post-placerholder2"
    />
  </Col>
</Row>
`;
